examples/jsm/loaders/LDrawLoader.js

```diff
@@ -1291,36 +1291,53 @@ class LDrawParsedCache {
 
 /////////////////////////////////////////////////////////////////
 
+// returns the material for an associated color code. If the color code is 16 for a face or 24 for
+// an edge then the passthroughColorCode is used.
+function getMaterialFromCode( colorCode, passthroughColorCode, materialHierarchy, forEdge ) {
 
-function getMaterialFromCode( colorCode, fallbackColorCode, materialHierarchy, forEdge ) {
+	const isPassthrough = ! forEdge && colorCode === '16' || forEdge && colorCode === '24';
+	if ( isPassthrough ) {
 
-	if ( ! forEdge && colorCode === '16' ) {
-
-		colorCode = fallbackColorCode;
-
-	}
-
-	if ( forEdge && colorCode === '24' ) {
-
-		colorCode = fallbackColorCode;
+		colorCode = passthroughColorCode;
 
 	}
 
 	return materialHierarchy[ colorCode ] || null;
 
 }
 
-function applyMaterialsToMesh( group, fallbackColorCode, materialHierarchy ) {
+// Applies the appropriate materials to a prebuilt hierarchy of geometry. Assumes that color codes are present
+// in the material array if they need to be filled in.
+function applyMaterialsToMesh( group, passthroughColorCode, materialHierarchy, finalMaterialPass = false ) {
 
+	// Returns the appropriate material for the object (line or face) given color code. If the code is "pass through"
+	// (24 for lines, 16 for edges) then the pass through color code is used. If that is also pass through then it's
+	// simply returned for the subsequent material application.
 	function getMaterial( c, colorCode ) {
 
-		if ( colorCode == '16' || colorCode === '24' ) {
+		const originalColorCode = colorCode;
+		const forEdge = c.isLineSegments || c.isConditionalLine;
+		const isPassthrough = ! forEdge && colorCode === '16' || forEdge && colorCode === '24';
+		if ( isPassthrough ) {
 
-			colorCode = fallbackColorCode;
+			colorCode = passthroughColorCode;
 
 		}
 
-		if ( colorCode === '16' || colorCode === '24' || ! ( colorCode in materialHierarchy ) ) {
+		// Only if processing the final pass of material applications do we want to use materials associated with
+		// the 24 and 16 color codes.
+		if ( ! finalMaterialPass ) {
+
+			const isPassthrough = ! forEdge && colorCode === '16' || forEdge && colorCode === '24';
+			if ( isPassthrough ) {
+
+				return originalColorCode;
+
+			}
+
+		}
+
+		if ( ! ( colorCode in materialHierarchy ) ) {
 
 			return colorCode;
```

Comment:
Shouldn't an error be generated here if ```finalMaterialPass === true```?

Reply:
Good suggestion! I just added a thrown error